api/src/main/java/org/pmiops/workbench/actionaudit/adapters/UserServiceAuditAdapterImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.pmiops.workbench.actionaudit.adapters;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.pmiops.workbench.actionaudit.ActionAuditEvent;
+import org.pmiops.workbench.actionaudit.ActionAuditService;
+import org.pmiops.workbench.actionaudit.ActionType;
+import org.pmiops.workbench.actionaudit.AgentType;
+import org.pmiops.workbench.actionaudit.TargetType;
+import org.pmiops.workbench.actionaudit.targetproperties.AccountTargetProperty;
+import org.pmiops.workbench.actionaudit.targetproperties.BypassTimeTargetProperty;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.model.DataAccessLevel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceAuditAdapterImpl implements UserServiceAuditAdapter {
+
+  private static final Logger log = Logger.getLogger(UserServiceAuditAdapterImpl.class.getName());
+  private final ActionAuditService actionAuditService;
+  private final Clock clock;
+  private Provider<DbUser> dbUserProvider;
+  private Provider<String> actionIdProvider;
+
+  @Autowired
+  public UserServiceAuditAdapterImpl(
+      ActionAuditService actionAuditService,
+      Clock clock,
+      Provider<DbUser> dbUserProvider,
+      @Qualifier("ACTION_ID") Provider<String> actionIdProvider) {
+    this.actionAuditService = actionAuditService;
+    this.clock = clock;
+    this.dbUserProvider = dbUserProvider;
+    this.actionIdProvider = actionIdProvider;
+  }
+
+  @Override
+  public void fireUpdateDataAccessAction(
+      DbUser targetUser, DataAccessLevel dataAccessLevel, DataAccessLevel previousDataAccessLevel) {
+    try {
+      ActionAuditEvent event =
+          new ActionAuditEvent(
+              clock.millis(),
+              AgentType.ADMINISTRATOR,
+              dbUserProvider.get().getUserId(),
+              dbUserProvider.get().getEmail(),
+              actionIdProvider.get(),
+              ActionType.EDIT,
+              TargetType.ACCOUNT,
+              AccountTargetProperty.REGISTRATION_STATUS.getPropertyName(),
+              targetUser.getUserId(),
+              previousDataAccessLevel.toString(),
+              dataAccessLevel.toString());
+      actionAuditService.send(event);
+    } catch (RuntimeException e) {
```

Comment:
It's a bit abnormal to catch a RuntimeException. What is the need for catching this specific type, and why is it necessary to log it at this level?

Reply:
Yes, it's abnormal. One overarching goal for this project is that if any code being instrumented fails or throws something, it won't be because of a problem with the auditing system. So the idea is to eat the exception and log an error. That way I'll still see a spike in errors and respond quickly, but I won't take down the UserService or WorkspacesController.

Reply:
It looks like the next level down already catches runtime exceptions. The only thing you'd feasibly be catching here is a programming error - is that the goal?

Reply:
Yeah, it's a bit belt-and-suspenders. I wanted a hermetic exception boundary at the level of the adapters, but you make a good point. Since the level below catches everything, we can drop this. Programming errors will show up soon enough.

Reply:
After removing the try/catch, I remembered one place where it's doing something for me. In the WorkspaceAuditAdapter I'm injecting a user provider, and there are situations [where it returns null](https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/actionaudit/adapters/WorkspaceAuditAdapterImpl.kt#L206), because we're not in a user-initiated query, or we're just starting up, or something.

It's better to return gracefully in that situation anyway than it is to throw an NPE.